### PR TITLE
[CBRD-23930] Change the TRUNCATE spec. with the CASCADE option

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/CBRD-23930

See the Jira issue for more information.

Note:
I'll add `cubrid.msg` files in other languages than en_US after this review is done.

### CODE REVIEW

File: msg/en_US/cubrid.msg [hunk 1]
```diff
@@ -1352,8 +1352,10 @@ Check the path of the key file (_keys) and make sure that it includes proper key
 1281 Error reserved for scalability development.
 1282 Error reserved for scalability development.
 1283 Error reserved for scalability development.
+1284 Cannot truncate the table due to the primary key referred to on %1$s. Try again with CASCADE option.
+1285 Cannot cascade truncate because a foreign key (%1$s) has not proper ON DELETE action.
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/en_US/cubrid.msg:1356]: Thanks

File: msg/en_US/cubrid.msg [hunk 2]
```diff
@@ -1352,8 +1352,10 @@ Check the path of the key file (_keys) and make sure that it includes proper key
 1281 Error reserved for scalability development.
 1282 Error reserved for scalability development.
 1283 Error reserved for scalability development.
+1284 Cannot truncate the table due to the pk is referred on %1$s. Try again with CASCADE option.
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/en_US/cubrid.msg]: already done :D

File: msg/en_US/cubrid.msg [hunk 3]
```diff
@@ -1352,8 +1352,10 @@ Check the path of the key file (_keys) and make sure that it includes proper key
...
+1285 Cannot cascade truncate because a foreign key (%1$s) has not proper ON DELETE action.
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/en_US/cubrid.msg]: OK

[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/en_US/cubrid.msg]: @hornetmj  How about now?

File: src/object/schema_manager.c [hunk 4]
```diff
@@ -15618,14 +15624,129 @@ sm_truncate_using_destroy_heap (MOP class_mop)
...
+  error = au_fetch_class (class_mop, &class_, AU_FETCH_READ, DB_AUTH_SELECT);
+  if (error != NO_ERROR || class_ == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      return er_errid ();
+    }
+
+  trun_classes.emplace (ws_oid (class_mop));
+
+  c = classobj_find_cons_primary_key (class_->constraints);
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply src/object/schema_manager.c:15652]: -> pk_constraint

File: src/object/schema_manager.c [hunk 5]
```diff
@@ -15618,14 +15624,129 @@ sm_truncate_using_destroy_heap (MOP class_mop)
...
+  if (c != NULL)		/* if no PK, it can be truncated */
+    {
+      for (fk_ref = c->fk_info; fk_ref; fk_ref = fk_ref->next)
+	{
+	  if (!is_cascade)
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply src/object/schema_manager.c]: With this I refactor the code a little.

File: src/object/schema_manager.c [hunk 6]
```diff
@@ -15618,14 +15624,134 @@ sm_truncate_using_destroy_heap (MOP class_mop)
...
+  if (error != NO_ERROR || class_ == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      return er_errid ();
+    }
+
+  trun_classes.emplace (ws_oid (class_mop));
+
+  pk_constraint = classobj_find_cons_primary_key (class_->constraints);
+  if (pk_constraint == NULL)
```
Comment (hornetmj) [inline src/object/schema_manager.c:15653]: I think it is more clear to add the following condition '|| pk_constraint->fk_info == NULL'.

File: src/object/schema_manager.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (hornetmj) [reply src/object/schema_manager.c]: good enough.

File: src/object/schema_manager.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (hornetmj) [reply src/object/schema_manager.c]: It is good for me. And I leave one more comment. Plz check it.

File: src/object/schema_manager.c (hunk 6 above)
Comment (wrlawodms) [reply src/object/schema_manager.c]: Good idea. I've revised it: [5fc8943](https://github.com/CUBRID/cubrid/pull/2738/commits/5fc8943a059f0dcd1bb34790198e6334d89c5ca0)

**Review by hornetmj — APPROVED** [review]

**Review by JoohoK — APPROVED** [review]

**Pull request merged**